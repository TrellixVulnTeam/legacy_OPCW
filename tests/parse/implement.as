# Add a `zero` method to `int`.
implement int { let zero(): int -> { return 0; } }

# Add a `slap` method to `int`.
implement int { let slap(self): int -> { return -self; } }
