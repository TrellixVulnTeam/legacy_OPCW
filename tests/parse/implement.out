ModuleDecl <?> _
  Nodes <?> 
    Implement <?>
      Ident <?> int
      Methods <?> 
        FuncDecl <?> zero
          Ident <?> int
          Type Parameters <?> 
          Parameters <?> 
          Block <?>
            Nodes <?> 
              ReturnExpr <?> 
                IntegerExpr <?> 0 (10)
    Implement <?>
      Ident <?> int
      Methods <?> 
        Member FuncDecl <?> slap
          Ident <?> int
          Type Parameters <?> 
          Parameters <?> 
          Block <?>
            Nodes <?> 
              ReturnExpr <?> 
                NumericNegateExpr <?>
                  Self <?> 
